pkg/git/git.go

```diff
@@ -80,8 +80,165 @@ func (h *stiGit) Clone(source, target string) error {
 	}
 	go pipeToLog(outReader, glog.Info)
 	go pipeToLog(errReader, glog.Error)
-	return h.runner.RunWithOptions(opts, "git", "clone", "--quiet", "--recursive", source, target)
+	//previously, we were doing the git clone with --quiet, but with the piping of the wrapping cmd stdout/stderr
+	// to glog on separate threads, we were causing a deadlock processing stdout/stderr channels across 4 goroutines.
+	// a SIGQUIT produced goroutine thread dump that shows the deadlock is included  after this method for historical
+	// purposes.
+	// Since it seems contradictory to run with --quiet and also pipe git output to glog, the workaround to the apparent
+	// go issue is the remove the --quiet option.  I also saved the core file produced by the SIGQUIT.
+	// a discussion on whether this is frequent enough or bad enough to open a Go issue on github could be warranted.
```

Comment:
hm.  I'm not seeing the actual deadlock/contention point here, I may need you to walk me through this on monday.  it seems like maybe we're just stuck waiting for input that never comes and we need to better handle/recognize when the output (stdout or stderr) stream is done?

and i'm not sure we want to remove --quiet since now we'll be spewing the output of git clone into the logs, right?  (which i guess isn't that bad, but it's less pretty anyway).

but regardless great find/investigation.  Not sure why you are the only one hitting this regularly, but i know @rcernich and @soltysh will be interested in this as they were trying to chase down git clone hangs previously.